Controller.java

```diff
@@ -7,58 +7,44 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/**
- * Servlet implementation class Controller
- */
+
 @WebServlet("/hello")
 public class Controller extends HttpServlet {
-
-	/**
-	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
-	 */
+	
+	protected void Writers (HttpServletRequest request, HttpServletResponse response, String fname) 
+			throws ServletException, IOException {
+		try (PrintWriter writer = response.getWriter()){
+			fname = request.getParameter("fname");
+			writer.println("nick: " + fname);
+	}
+}
 	protected void doGet(HttpServletRequest request, HttpServletResponse response) 
             throws ServletException, IOException {    
-
-        try (PrintWriter writer = response.getWriter()){
-            String nick = request.getParameter("nick");
-        	writer.println("Nickname: " + nick);
-        }
+        Writers(request, response, request.getParameter("nick: "));
```

Comment:
It is a bad idea to do some actions with data when calls a method. Move request.getParameter... before method call